vm/compiler/src/program/instruction/operation/macros.rs

```diff
@@ -710,14 +710,36 @@ mod tests {
 
                     // Check the operation on randomly-sampled values.
                     for i in 0..num_iterations {
+                        macro_rules! sample_value {
+                            (I8, I8) => { sample_value!(I128, I128) };
+                            (I16, I16) => { sample_value!(I128, I128) };
+                            (I32, I32) => { sample_value!(I128, I128) };
+                            (I64, I64) => { sample_value!(I128, I128) };
```

Comment:
nit: these branches can probably be compacted to `(I8, I8) | (I16, I16) | ... => { sample_value!(I128, I128) };`

Reply:
It doesn't seem to work.
The compiler errors with `error: no rules expected the token \`|\``

Reply:
ah true, I thought I saw a `match` there, but these are macro branches, so this doesn't apply :+1: